src/ngx_http_lua_api.c

```diff
@@ -194,13 +198,12 @@ ngx_http_lua_shared_memory_init(ngx_shm_zone_t *shm_zone, void *data)
 
     lmcf->shm_zones_inited++;
 
-    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts
-        && lmcf->init_handler)
+    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts)
     {
         saved_cycle = ngx_cycle;
         ngx_cycle = ctx->cycle;
 
-        rc = lmcf->init_handler(ctx->log, lmcf, lmcf->lua);
+        rc = ngx_http_lua_delay_init_handler();
```

Comment:
When there is no `lua_shared_dict` directive configured in `nginx.conf`, there the delayed execution will never run, am I correct? This looks wrong to me.